### DISCUSSION TITLE
How do I change the default workspace for go-agent

### DISCUSSION THREAD

**Discussion by xlw712** [gocd/gocd#11370] [Q&A] [answered]
I used the plugin and when I started the go-agent pod, the default workspace I looked at was  `/godata/pipeline/`, but because of the special restrictions on directories in my jobs, directories starting with `p` must not be allowed.Is there a way to set the default directory?

**Comment by xlw712** [comment, marked as answer]
@chadlwilson Can it be configured?

**Reply by chadlwilson** [reply]
For agents, you cannot change the name of the `pipelines` subdirectory - you can only change the whole `/godata` folder using the `VOLUME_DIR` environment variable as noted within docker hub pages for the various agent images: e.g https://hub.docker.com/r/gocd/gocd-agent-ubuntu-20.04 - so the workspace would still be within `/my-other-folder/somewhere/pipelines`.

As a general principle, nothing in your jobs/tasks should be sensitive to where the agent decides to store its working information, especially not the naming of the folder.

**Reply by xlw712** [reply]
OK,Thanks 
Only I can fit in.
